Extensions/timestamps.js

```diff
@@ -224,12 +253,88 @@ XKit.extensions.timestamps = new Object({
 
 			var timestamp = responseData.posts[0].timestamp;
 			date_element.html(this.format_date(moment(new Date(timestamp * 1000))));
-			date_element.removeClass("xtimestamp_loading");
+			date_element.removeClass("xtimestamp-loading");
 			XKit.storage.set("timestamps", "xkit_timestamp_cache_" + post_id, timestamp);
 		})
 		.catch(() => this.show_failed(date_element));
 	},
 
+	react_add_timestamps: function() {
+		var posts = $("[data-id]:not(.xkit_timestamps)");
+
+		if (posts.length === 0) {
+			return;
+		}
+
+		XKit.extensions.timestamps.check_quota();
+
+		posts.each(function() {
+			var post = $(this);
+			post.addClass("xkit_timestamps");
+
+			var post_id = $(this).attr("data-id");
```

Comment:
`$(this)` is already saved as `post`; this is significant because `$` is a function, so it's best to reduce calls to it where possible for efficiency

```suggestion
			var post_id = post.attr("data-id");
```